### ISSUE TITLE
Windows has high memory usage

### ISSUE THREAD

**Issue by jonathandturner** [servo/servo#15572]
As mentioned [here](https://github.com/servo/servo/issues/12125#issuecomment-277797098), [here](https://github.com/servo/servo/issues/12125#issuecomment-277800514), and [here](https://github.com/servo/servo/issues/12125#issuecomment-277873838) the Windows build is also leaking a fair amount of memory.

This may be related to #15428.

cc @codec-abc, @leeoniya, @BladeMight who mentioned experiencing the issue

**Comment by codec-abc** [comment]
Could you confirm that the link to download the last Windows version for servo is :
https://download.servo.org/nightly/windows-msvc/servo-latest.msi

**Comment by leeoniya** [comment]
yes this is still an issue. i could not get any site on the start page to display at all. instead servo chews through RAM and has to be process-killed to fully exit.

**Comment by codec-abc** [comment]
Tested again today with latest nightly. Seems like the problem is gone.

**Comment by PEmu1** [comment]
I have not been able to reproduce this issue in the latest nightly builds either. Previously Servo would use up many GB of memory until it needed to be killed where now I find it stays under 1 GB with multiple tabs open.

**Comment by jdm** [comment]
Seems like this can be closed, in that case. Thanks for checking!

**Comment by jonathandturner** [comment]
@jdm - this still repros.  With last night's nightly, if you click on the link to go to https://medium.com/@dourvaris/my-2015-macbook-pro-retina-exploded-119ea5ea9d1f#.pgd3f8jme from hacker news, you get the same high memory behaviour. It seems to not be as common now, though.

**Comment by PEmu1** [comment]
I followed the steps in @jonathandturner's post and was able to reproduce the issue as well. Seems to not be fixed after all.

**Comment by codec-abc** [comment]
My bad for the wrong reporting. I confirm that the issue is still here but is triggered quite less frequently than before.

**Comment by codec-abc** [comment]
I can still reproduce but with the nightly and compiling myself servo with MSVC. Note that the memory leak only happen in release. In debug for the given medium page I am at a stable ~200MB Memory usage.

**Comment by jonathandturner** [comment]
Just a bit of follow-up, to echo what @codec-abc mentioned.

The debug build doesn't finish rendering the page correctly.  The image at the top fails to load, showing this:

<img width="583" alt="servo_debug_exploded" src="https://cloud.githubusercontent.com/assets/547158/23882714/531f6722-08c7-11e7-9053-8133582841a1.png">

Debug mode holds steady between 200 and 300 MB.  Memory climbs towards 300 then drops down to 200 or so again.

Release mode is able to render the image in the middle:

<img width="579" alt="servo_release_exploded" src="https://cloud.githubusercontent.com/assets/547158/23882708/48ebd1d2-08c7-11e7-86ae-478620a5ce5f.png">

But, it quickly begins to consume memory without end.

I attached a debugger and was able to see a little glimpse of what is going on, but would still need to dig in further to see if this is related or just a red herring:

<img width="508" alt="large_mem_usage_servo" src="https://cloud.githubusercontent.com/assets/547158/23882766/a221c1d0-08c7-11e7-8ce4-b849e74444a0.png">
